plugins/org.eclipse.fordiac.ide.debug.ui/src/org/eclipse/fordiac/ide/debug/ui/fb/FBLaunchConfigurationTab.java

```diff
@@ -138,6 +235,29 @@ public void performApply(final ILaunchConfigurationWorkingCopy configuration) {
 		configuration.setAttribute(FBLaunchConfigurationAttributes.REPEAT_EVENT, repeatEventCheckbox.getSelection());
 		configuration.setAttribute(FBLaunchConfigurationAttributes.KEEP_RUNNING_WHEN_IDLE,
 				keepDebuggerRunningCheckbox.getSelection());
+
+		if (systemTimeRadio.getSelection()) {
+			configuration.setAttribute(FBLaunchConfigurationAttributes.CLOCK_MODE,
+					FBLaunchConfigurationAttributes.SYSTEM);
+		}
+		if (incrementTimeRadio.getSelection()) {
+			configuration.setAttribute(FBLaunchConfigurationAttributes.CLOCK_MODE,
+					FBLaunchConfigurationAttributes.INCREMENT);
+		}
+		if (manualTimeRadio.getSelection()) {
+			configuration.setAttribute(FBLaunchConfigurationAttributes.CLOCK_MODE,
+					FBLaunchConfigurationAttributes.MANUAL);
+		}
+		try {
+			final String debug_time = debugTimeText.getText();
+			if (debug_time != null) {
+				configuration.setAttribute(FBLaunchConfigurationAttributes.CLOCK_INTERVAL, debug_time);
+			} else {
+				configuration.setAttribute(FBLaunchConfigurationAttributes.CLOCK_INTERVAL, "0"); //$NON-NLS-1$
+			}
```

Comment:
I believe `Text.getText()` never returns `null`? I would also suggest fixing the variable name to conform to the camel-case naming convention.

Reply:
fxed